src/Microsoft.Graph.Core/Requests/HttpProvider.cs

```diff
@@ -29,9 +27,10 @@ public class HttpProvider : IHttpProvider
         /// <summary>
         /// Constructs a new <see cref="HttpProvider"/>.
         /// </summary>
+        /// <param name="authenticationProvider">The <see cref="IAuthenticationProvider"/> for authenticating request messages.</param>
         /// <param name="serializer">A serializer for serializing and deserializing JSON objects.</param>
-        public HttpProvider(ISerializer serializer = null)
```

Comment:
This is a breaking change.  I think we are going to need to add a new constructor, or also make the authenticationProvider parameter optional.